packages/core/src/routes/experience/classes/verifications/password-verification.ts

```diff
@@ -0,0 +1,94 @@
+import {
+  VerificationType,
+  interactionIdentifierGuard,
+  type InteractionIdentifier,
+} from '@logto/schemas';
+import { type ToZodObject } from '@logto/schemas/lib/utils/zod.js';
+import { generateStandardId } from '@logto/shared';
+import { z } from 'zod';
+
+import RequestError from '#src/errors/RequestError/index.js';
+import type Libraries from '#src/tenants/Libraries.js';
+import type Queries from '#src/tenants/Queries.js';
+import assertThat from '#src/utils/assert-that.js';
+
+import { findUserByIdentifier } from '../../utils.js';
+
+import { type Verification } from './verification.js';
+
+export type PasswordVerificationRecordData = {
+  id: string;
+  type: VerificationType.Password;
+  identifier: InteractionIdentifier;
+  /* The userId of the user that has been verified */
+  userId?: string;
```

Comment:
```suggestion
  verifiedUserId?: string;
```

Reply:
will be updated in the following PR, once the `verifiedUserId` getter method is removed.